dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/alert/AlertClientService.java

```diff
@@ -78,40 +72,46 @@ public void close() {
 
     /**
      * alert sync send data
-     * @param groupId
-     * @param title
-     * @param content
-     * @return
      */
-    public AlertSendResponseCommand sendAlert(int groupId, String title,  String content) {
-        return this.sendAlert(this.host,this.port,groupId,title,content);
+    public AlertSendResponseCommand sendAlert(int groupId, String title, String content) {
+        return this.sendAlert(this.host, this.port, groupId, title, content);
     }
 
     /**
      * alert sync send data
+     *
      * @param host host
      * @param port port
      * @param groupId groupId
      * @param title title
      * @param content content
      * @return AlertSendResponseCommand
      */
-    public AlertSendResponseCommand sendAlert(String host, int port, int groupId, String title,  String content) {
+    public AlertSendResponseCommand sendAlert(String host, int port, int groupId, String title, String content) {
         logger.info("sync alert send, host : {}, port : {}, groupId : {}, title : {} ", host, port, groupId, title);
         AlertSendRequestCommand request = new AlertSendRequestCommand(groupId, title, content);
-        final Host address = new Host(host, port);
-        try {
-            Command command = request.convert2Command();
-            Command response = this.client.sendSync(address, command, ALERT_REQUEST_TIMEOUT);
-            if (response != null) {
-                return JsonSerializer.deserialize(response.getBody(), AlertSendResponseCommand.class);
+        AlertSendResponseCommand alertSendResponseCommand = null;
+        String[] hosts = host.split(",");
+        for (String hostName : hosts) {
+            final Host address = new Host(hostName, port);
```

Comment:
1. Split the host by comma. If the alert. Listen. Host =IP1,IP2 in Worker.properties, will the SQLTask alert be sent twice?
2.Alert-server has already used ZK, this time it is recommended that the alert-server be registered in ZK, synchronous Alert get alert-server host from ZK (like worker/master)

---
1.host按逗号分割，如果worker.properties内alert.listen.host=IP1,IP2, 那是不是sqlTask的告警会发送两遍？
2.Alert-server 已经使用ZK，此次建议alert-server在zk注册，同步告警从zk获取alert-server host（像worker/master一样）

Reply:
1.  it will not resend twice, the return will drop on the first success.
2.Second, I'll do a unified optimization after that. Because now there are duplicate alerts if the worker hangs up. The worker will immediately hang up, and the worker will send an alert of his hanging up.
---


1. 不会重发两次，第一次成功就return掉了。
2.第二点，之后我会统一优化一次。因为现在如果worker挂掉，存在重复告警。存在worker马上挂掉，自己也会自己发自己挂掉的告警。